addons/mass_mailing/models/mass_mailing.py

```diff
@@ -80,102 +64,134 @@ class MassMailingContact(osv.Model):
-        rec_id = self.create(cr, uid, {'name': name, 'email': email}, context=context)
-        return self.name_get(cr, uid, [rec_id], context)[0]
+    def name_create(self, name):
+        name, email = self.get_name_email(name)
+        contact = self.create({'name': name, 'email': email})
+        return contact.display_name
 
-    def add_to_list(self, cr, uid, name, list_id, context=None):
-        name, email = self.get_name_email(name, context=context)
-        rec_id = self.create(cr, uid, {'name': name, 'email': email, 'list_id': list_id}, context=context)
-        return self.name_get(cr, uid, [rec_id], context)[0]
+    @api.model
+    def add_to_list(self, name, list_id):
+        name, email = self.get_name_email(name)
+        contact = self.create({'name': name, 'email': email, 'list_id': list_id})
+        return contact.display_name
 
-    def message_get_default_recipients(self, cr, uid, ids, context=None):
-        res = {}
-        for record in self.browse(cr, uid, ids, context=context):
-            res[record.id] = {'partner_ids': [], 'email_to': record.email, 'email_cc': False}
-        return res
+    def message_get_default_recipients(self):
+        return dict((record.id, {'partner_ids': [], 'email_to': record.email, 'email_cc': False}) for record in self)
 
-    def message_receive_bounce(self, cr, uid, ids, mail_id=None, context=None):
+    def message_receive_bounce(self, mail_id=None):
         """Called by ``message_process`` when a bounce email (such as Undelivered
         Mail Returned to Sender) is received for an existing thread. As contacts
         do not inherit form mail.thread, we have to define this method to be able
         to track bounces (see mail.thread for more details). """
-        for obj in self.browse(cr, uid, ids, context=context):
-            self.write(cr, uid, [obj.id], {'message_bounce': obj.message_bounce + 1}, context=context)
+        for contact in self:
+            contact.message_bounce = contact.message_bounce + 1
+
 
+class MassMailingStage(models.Model):
 
-class MassMailingStage(osv.Model):
     """Stage for mass mailing campaigns. """
     _name = 'mail.mass_mailing.stage'
     _description = 'Mass Mailing Campaign Stage'
     _order = 'sequence'
 
-    _columns = {
-        'name': fields.char('Name', required=True, translate=True),
-        'sequence': fields.integer('Sequence'),
-    }
+    name = fields.Char(required=True, translate=True)
+    sequence = fields.Integer()
 
-    _defaults = {
-        'sequence': 0,
-    }
 
-
-class MassMailingCampaign(osv.Model):
+class MassMailingCampaign(models.Model):
     """Model of mass mailing campaigns. """
     _name = "mail.mass_mailing.campaign"
     _description = 'Mass Mailing Campaign'
     _rec_name = "campaign_id"
     _inherits = {'utm.campaign': 'campaign_id'}
 
-    def _get_statistics(self, cr, uid, ids, name, arg, context=None):
+    name = fields.Char(required=True)
```

Comment:
This field name seems to come from a bad rebase. I think it has been removed as the _rec_name is now the campaign id